pkg/identity/cidr/identity.go

```diff
@@ -61,7 +61,7 @@ func AllocateCIDRIdentities(prefixes []*net.IPNet) (res []*identity.Identity, er
 			res = nil
 			break
 		}
-		res = append(res, id)
+		res[i] = id
```

Comment:
For my own education, is there a functional change here?

My understanding of the prior code is that it would create a slice that has the capacity of the length of the prefixes, but with 0 elements inside, then the `append()` would not reallocate. The new code just creates a slice with the length of the prefixes and populates each element.

Reply:
The problematic part is the `continue` above which will cause the `identities` and `prefixes` to not map 1-1 if prefixes contain nil entries. I'm not entirely sure when nil entries happen but the if condition makes me believe that it is a valid case.